pkg/generate/app/cmd/resolve.go

```diff
@@ -502,6 +503,11 @@ func AddMissingComponentsToRefBuilder(
 				errs = append(errs, fmt.Errorf("the Dockerfile in the repository %q has no FROM instruction", info.Path))
 				continue
 			}
+			err := exposedPortsAreNumeric(dockerfileutil.LastExposedPorts(node))
+			if err != nil {
+				errs = append(errs, fmt.Errorf("the Dockerfile in the repository has invalid EXPOSE instruction: %v", err))
```

Comment:
s/the Dockerfile in the repository/the Dockerfile/
s/has invalid/has an invalid/